Source/WebKit/UIProcess/WebAuthentication/Cocoa/WebAuthenticatorCoordinatorProxy.mm

```diff
@@ -320,13 +320,24 @@ static inline ASAuthorizationPublicKeyCredentialLargeBlobSupportRequirement toAS
     return requests;
 }
 
-void WebAuthenticatorCoordinatorProxy::pauseConditionalAssertion()
+WeakPtr<WebAuthenticatorCoordinatorProxy>& WebAuthenticatorCoordinatorProxy::activeConditionalMediationProxy()
 {
-    if (m_paused)
+    static MainThreadNeverDestroyed<WeakPtr<WebAuthenticatorCoordinatorProxy>> proxy;
+    return proxy.get();
+}
+
+void WebAuthenticatorCoordinatorProxy::pauseConditionalAssertion(CompletionHandler<void()>&& completionHandler)
+{
+    if (m_paused) {
+        completionHandler();
         return;
+    }
     m_paused = true;
-    if (m_isConditionalMediation)
+    if (m_isConditionalMediation) {
+        m_cancelHandler = WTFMove(completionHandler);
```

Comment:
What if something is waiting for `m_cancelHandler` to be called? Should we trigger it before reassigning it?

Reply:
It looks like in `WebAuthenticatorCoordinatorProxy::cancel` when we pass a new handler we just overwrite any cancel handler already in existence, so perhaps no special handling is needed here.